src/gridfs.rs

```diff
@@ -322,3 +230,9 @@ impl GridFsBucket {
         Ok(())
     }
 }
+
+impl Error {
+    fn into_futures_io_error(self) -> futures_io::Error {
+        futures_io::Error::new(futures_io::ErrorKind::Other, self)
```

Comment:
A `futures_io` error is just a `std::io::Error`, which we wrap in our `ErrorKind` too. In the event that our `Error` is that variant, we should just return it straight without mapping to the `Other` case.

Reply:
Ohh right, forgot about that issue. We could try to deal with this via [`Arc::try_unwrap`](https://doc.rust-lang.org/std/sync/struct.Arc.html#method.try_unwrap), but it's probably not worth it. Seems fine to leave this as-is.